### ISSUE TITLE
Firewall /  iptables

### ISSUE THREAD

**Issue by smossber** [redhatNORDICS/consulting-lab#4]
I think we need some form over overlaying framework (or whatever we want to call it) on how we deal with iptables and networks overall.

Examples of stuff we want to be able to do
- limit and open up the virtual networks independently.
- a mgmt network should be accessible from all or many other networks on specific ports
- most networks can't reach the internet, but has to use a proxy
- VPN users should be able to ssh to all networks? 
- Secure the lab, so it's only reachable by VPN or port knocking. 

How do do this in a managable way?

**Comment by smossber** [comment]
I've been tinkering with how to best do locked down networks.

libvirt offers out of the box 3 different network modes, NAT'ed, isolated and routed.
https://wiki.libvirt.org/page/VirtualNetworking

Bedsides putting up and managing the virtual bridges for you it will also insert rules to the different iptables chains.
I find the difference between isolated and routed comes down to differences in the FORWARD table.


This doc shows the difference and also discuss firewalling which should be interesting to us: https://libvirt.org/firewall.html 
However, the solution offered there didn't quite ring with me. 

Anyway, the diff in isolated and routed is that isolated REJECT anything that is routed from another interface to the networks virtual interface.
So we have networkA with virbrA interface. Any traffic going to (output) virbrA from any other interface (input) is REJECTED.
It also works the same way in the opposite direction, i.e. anything coming in on virbrA that is not destined for virbrA will be rejected. 

Routed however, allows anything going into or out from the virtual interface with the exception that it's limited to the the subnet range.

So in short, we wither get wide open forwarding between the networks.. or completely locked down.
And we want something in between.. and we have to decide from which direction we want to do it from. 
The goal being:
Pretty locked down networks on creation but with some standard openings, and it should be easy to make modifications and get an overview.

**Comment by smossber** [comment]
On  a second thought, maybe the nwfilter is the way to go? 
https://libvirt.org/formatnwfilter.html#goals
Haven't tested it.

**Comment by smossber** [comment]
Merged https://github.com/redhatNORDICS/consulting-lab/pull/12
